### PR TITLE
feature/UN-37 hooks, rebase and best-practices rules

### PR DESCRIPTION
**Pull request by luispdm**

### CODE REVIEW

File: git/pre-commit [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+go mod tidy
+
+#### Check all files for suspicious constructs
+echo "${BGBLUE}>> go vet${NC}"
+go vet ${FILES}
+check_errors $?
+
+#### Launching tests
+echo "${BGBLUE}>> go test -count=1 -race ./...${NC}"
+go test -count=1 -race ./...
```
Comment (zorjak) [inline git/pre-commit:68]: does this mean that we have to run all the tests before committing? Maybe it is too hard if there are many tests?

Comment (luispdm) [reply git/pre-commit:68]: Yes, that runs all tests. My idea is to run just unit tests in the pre-commit hook (it's all we have for now), so it shouldn't take more than a few seconds. This way you'd know if your changes have broken some tests and you wouldn't have to wait for the CI to tell you

File: README.md [hunk 2]
```diff
@@ -24,4 +24,35 @@ Right now there is only one method supported:
...
+### Best-practices
+Follow best-practices and guidelines published by Google and the golang open-source community, such as:
+
+Effective Go - https://go.dev/doc/effective_go
+
+Go style guide - https://google.github.io/styleguide/go/
+
+Go code review comments - https://github.com/golang/go/wiki/CodeReviewComments
+
+Go test comments - https://github.com/golang/go/wiki/TestComments
```
Comment (kudlaj) [inline README.md]: I think we should discuss this before we add this

File: git/pre-commit (hunk 1 above)
Comment (AlexRipoll) [reply git/pre-commit:68]: I thinks it's fine to do it this way, if changes to a file breaks some unit test tests, you will need to update them anyway

File: README.md (hunk 2 above)
Comment (luispdm) [reply README.md]: removed, as agreed

**Review by kudlaj — APPROVED** [review]

**Review by AlexRipoll — APPROVED** [review]

**Pull request merged**